python_modules/dagster/dagster/core/definitions/decorators/solid.py

```diff
@@ -46,6 +47,12 @@ def positional_inputs(self) -> List[str]:
         input_args = params[1:] if self.has_context_arg() else params
         return positional_arg_name_list(input_args)
 
+    @lru_cache(maxsize=1)
```

Comment:
nit: maybe add a memoized `_get_function_params` instead of each of these getters. Thats the only expensive part right?